project/Npm.scala

```diff
@@ -0,0 +1,30 @@
+import sbt.{Command, File}
+
+import scala.sys.process.{Process, ProcessBuilder}
+
+object Npm {
+
+  def npmCommand(base: File) = Command.args("npm", "<npm-command>") { (state, args) =>
+    process(base, args:_*) !;
+    state
+  }
+
+  def npmProcess(failureMessage: String)(base: File, args: String*): Int = {
+    val processBuilder = process(base, args: _*)
+    val exitValue = processBuilder.run().exitValue()
+    if (exitValue != 0) {
+      throw new Exception(failureMessage)
+    } else exitValue
+  }
+
+  private def process(base: File, args: String*): ProcessBuilder = {
+    // this hasn't actually been tested on windows, so it might not work :)
```

Comment:
If it hasn't been tested on Windows I'd suggest erroring on Windows environments - it makes it easier to chase down the problem.

Reply:
I copied it from cato-frontend, so I'm assuming it does actually work - its just a caveat.